gtda/images/filtrations.py

```diff
@@ -98,18 +108,17 @@ def fit(self, X, y=None):
 
         """
         X = check_array(X, allow_nd=True)
-
-        self.n_dimensions_ = len(X.shape) - 1
+        self.n_dimensions_ = X.ndim - 1
+        if (self.n_dimensions_ < 2) or (self.n_dimensions_ > 3):
+            warn(f"Input of `fit` contains arrays of dimension "
+                 f"{self.n_dimensions_}.")
```

Comment:
I've tried to avoid abusing `validate_params` to turn it into a validation for input (i.e. `X`), so this is a conceptual decision to make the intent of `validate_params` well-defined, and not a technical one. What do you think? E.g. would this not be covered by the solution for #318?